doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/test-scenarios-execution-rest-api-proc.adoc

```diff
@@ -0,0 +1,516 @@
+[id='test-scenarios-execution-rest-api-proc']
+
+= Executing a test scenario using the {KIE_SERVER} REST API
+
+Directly interacting with the REST endpoints of {KIE_SERVER} provides the most separation between the calling code and the decision logic definition.
+You can use the {KIE_SERVER} REST API to execute the test scenarios externally. It executes the test scenarios against the deployed project.
+
+[NOTE]
+====
+This functionality is disabled by default, use `org.kie.scenariosimulation.server.ext.disabled` system property to enable it.
+====
+
+For more information about the {KIE_SERVER} REST API, see
+ifdef::DM,PAM[]
+{URL_KIE_APIS}[_{KIE_APIS}_].
+endif::[]
+ifdef::DROOLS,JBPM,OP[]
+xref:kie-server-rest-api-con_kie-apis[].
+endif::[]
+
+.Prerequisites
+
+* {KIE_SERVER} is installed and configured, including a known user name and credentials for a user with the `kie-server` role. For installation options, see
+ifdef::DM,PAM[]
+{URL_PLANNING_INSTALL}[_{PLANNING_INSTALL}_].
+endif::[]
+ifdef::DROOLS,JBPM,OP[]
+<<_installationandsetup>>.
+endif::[]
+
+* You have built the test scenario project as a KJAR artifact and deployed it to {KIE_SERVER}.
+* You have the ID of the KIE container.
+
+.Procedure
+
+. Determine the base URL for accessing the {KIE_SERVER} REST API endpoints. This requires knowing the following values (with the default local deployment values as an example):
++
+--
+* Host (`localhost`)
+* Port (`8080`)
+* Root context (`kie-server`)
+* Base REST path (`services/rest/`)
+
+Example base URL in local deployment for the traffic violations project:
+
+`\http://localhost:8080/kie-server/services/rest/server/containers/traffic_1.0.0-SNAPSHOT`
+--
+
+. Determine user authentication requirements.
++
+When users are defined directly in the {KIE_SERVER} configuration, HTTP Basic authentication is used and requires the user name and password. Successful requests require that the user have the `kie-server` role.
++
+The following example demonstrates how to add credentials to a curl request:
++
+[source]
+----
+curl -u username:password <request>
+----
++
+If {KIE_SERVER} is configured with Red Hat Single Sign-On, the request must include a bearer token:
++
+[source,java]
+----
+curl -H "Authorization: bearer $TOKEN" <request>
+----
+
+. Specify the format of the request and response. The REST API endpoints work with XML format and are set using request headers:
++
+.XML
+[source]
+----
+curl -H "accept: application/xml" -H "content-type: application/xml"
+----
+
+. Execute the test scenario:
++
+*[POST]* `server/containers/{containerId}/scesim`
++
+Example curl request:
++
+[source]
+----
+curl -u 'testadmin:admin1234;' -H "accept: application/xml" -X POST "http://localhost:8080/kie-server/services/rest/server/containers/traffic_1.0.0-SNAPSHOT/scesim" --data '<ScenarioSimulationModel version="1.5">
```

Comment:
--data '<ScenarioSimulationModel version="1.5"> 
This request is not finished. You need ether add a place holder for xml or include content directly into.

Reply:
@dupliaka, I did not get it? Can you please explain it to me?

Reply:
![image](https://user-images.githubusercontent.com/21653197/76405246-21ae6200-6388-11ea-9e22-2b0ae183bc5e.png)
So you mention example url, but in data you put "'<ScenarioSimulationModel version="1.5">"
That is a part of input scesim file. This would obviously not work. So my point is that body is wrong and if you want to tell the user that he need to pass an scenario file directly into body than you need to mention it and add some reference/ place holder to let him know that the scesm should be there